File: CS/FixedRowExample/ViewModel/TaskViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Input;

namespace FixedRowExample
{
    public class TaskViewModel
    {

        private ObservableCollection<Task> _TaskData;

        public ObservableCollection<Task> TaskData
        {
            get
            {
                if (_TaskData == null)
                {
                    _TaskData = new ObservableCollection<Task>();
                    for (int i = 0; i < 300; i++)
                    {
                        _TaskData.Add(new Task() { Name = "Task " + i, Number = i, Date = new DateTime(2014, 10, new Random().Next(1, 31)), IsCompleted = i % 2 != 0 });
                    }
                }
                return _TaskData;
            }
        }
    }

    public class Task : INotifyPropertyChanged
    {
        private string _Name;
        private int _Number;
        private DateTime _Date;
        private bool _IsCompleted;

        public event PropertyChangedEventHandler PropertyChanged;

        private void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

        public string Name
        {
            get
            {
                return _Name;
            }

            set
            {
                _Name = value;
                OnPropertyChanged("Name");
            }
        }

        public int Number
        {
            get
            {
                return _Number;
            }
            set
            {
                _Number = value;
                OnPropertyChanged("Number");
            }
        }

        public DateTime Date
        {
            get
            {
                return _Date;
            }
            set
            {
                _Date = value;
                OnPropertyChanged("Date");
            }
        }

        public bool IsCompleted
        {
            get
            {
                return _IsCompleted;
            }
            set
            {
                _IsCompleted = value;
                OnPropertyChanged("IsCompleted");
            }
        }
    }
}

File: VB/FixedRowExample/ViewModel/TaskViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows
Imports System.Windows.Input

Namespace FixedRowExample
	Public Class TaskViewModel

		Private _TaskData As ObservableCollection(Of Task)

		Public ReadOnly Property TaskData() As ObservableCollection(Of Task)
			Get
				If _TaskData Is Nothing Then
					_TaskData = New ObservableCollection(Of Task)()
					For i As Integer = 0 To 29
						_TaskData.Add(New Task() With {.Name = "Task " & i, .Number = i, .Date = New DateTime(2014, 10, New Random().Next(1, 31)), .IsCompleted = i Mod 2 <> 0})
					Next i
				End If
				Return _TaskData
			End Get
		End Property
	End Class

	Public Class Task
		Implements INotifyPropertyChanged
		Private _Name As String
		Private _Number As Integer
		Private _Date As DateTime
		Private _IsCompleted As Boolean

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Private Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub

		Public Property Name() As String
			Get
				Return _Name
			End Get

			Set(ByVal value As String)
				_Name = value
				OnPropertyChanged("Name")
			End Set
		End Property

		Public Property Number() As Integer
			Get
				Return _Number
			End Get
			Set(ByVal value As Integer)
				_Number = value
				OnPropertyChanged("Number")
			End Set
		End Property

		Public Property [Date]() As DateTime
			Get
				Return _Date
			End Get
			Set(ByVal value As DateTime)
				_Date = value
				OnPropertyChanged("Date")
			End Set
		End Property

		Public Property IsCompleted() As Boolean
			Get
				Return _IsCompleted
			End Get
			Set(ByVal value As Boolean)
				_IsCompleted = value
				OnPropertyChanged("IsCompleted")
			End Set
		End Property
	End Class
End Namespace
